### ISSUE TITLE
Package dependent code loading doesn't work in v0.7 ?

### ISSUE THREAD

**Issue by jonathanBieler** [GiovineItalia/Compose.jl#308]
I'm trying to update Immerse to v0.7 and I'm getting an issue with this part:

https://github.com/GiovineItalia/Compose.jl/blob/master/src/Compose.jl#L167

This mechanism seems to work fine in the REPL:

```
julia> using Cairo

julia> using Compose
[ Info: Loading Cairo backend into Compose.jl

julia> Compose.absolute_native_units
absolute_native_units (generic function with 4 methods)


julia> module A
           using Cairo
           using Compose
           
           import Compose.absolute_native_units            
       end
[ Info: Loading Cairo backend into Compose.jl
Main.A
```

But if I put the code above in a package, it fails to import absolute_native_units:

```
julia> using TestCompose
[ Info: Recompiling stale cache file /Users/jbieler/.julia/compiled/v0.7/TestCompose/Tl9GO.ji for TestCompose [c558b20d-ff79-52a3-ae8b-9e51a808e850]
┌ Warning: Package TestCompose does not have Compose in its dependencies:
│ - If you have TestCompose checked out for development and have
│   added Compose as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with TestCompose
└ Loading Compose into TestCompose from project dependency, future warnings for TestCompose are suppressed.
WARNING: could not import Compose.absolute_native_units into TestCompose
[ Info: Loading Cairo backend into Compose.jl
```

Any way to fix this ? 

Might be related to https://github.com/GiovineItalia/Compose.jl/issues/250, but I'm not sure what's the status of it.

**Comment by tlnagy** [comment]
Do you have `Compose` in `TestCompose`'s REQUIRE file?

**Comment by jonathanBieler** [comment]
Yes, I also tried to add them in a toml file but I can't get rid of this warning, I'm not sure what's going on with that. Do you think that's the cause of the issue ?

**Comment by bjarthur** [comment]
might try reversing the order of the `using`s.  Compose before Cairo.

**Comment by jonathanBieler** [comment]
That doesn't seem to make a difference, I'v put a couple of prints in the packages to try to understand what's going on. Compose init gets called twice but the first one doesn't load Cairo:

```
julia> using TestCompose
[ Info: Recompiling stale cache file /Users/jbieler/.julia/compiled/v0.7/TestCompose/Tl9GO.ji for TestCompose [c558b20d-ff79-52a3-ae8b-9e51a808e850]

"Compose init called" = "init called"
"TestCompose trying to import absolute_native_units" = "trying to import absolute_native_units"
WARNING: could not import Compose.absolute_native_units into TestCompose

"Compose init called" = "init called"
[ Info: Loading Cairo backend into Compose.jl
```

**Comment by tlnagy** [comment]
> "Compose init called" 

Did you add the print statement to this function?
 https://github.com/GiovineItalia/Compose.jl/blob/69b4d9590e9c2f843480b31be8571ca1e26126b5/src/Compose.jl#L166-L169

This might be an issue with `Requires.jl` which we depend on for conditional loading of Cairo. In the init function we just load code that will get loaded when Cairo is loaded. This works fine in interactive environments, but I'm not sure if it works in packages that load both Cairo and Compose (your case).

**Comment by jonathanBieler** [comment]
>Did you add the print statement to this function?

Yes. I think it's a problem of the sort indeed. I'll check if I can reproduce a simpler example that uses Requires.

**Comment by jonathanBieler** [comment]
I managed to make it work by calling `link_cairo` explicitly in Immerse:

```julia
@eval Compose begin import Cairo end
Compose.link_cairo()
```

I had to do a couple of [changes](https://github.com/GiovineItalia/Compose.jl/pull/313).

**Comment by bjarthur** [comment]
@jonathanBieler might this problem with Immerse.jl be related to the scope of module search path mentioned in the Requires.jl README?

> In the @require block, or any included files, you can use or import the package, but note that you must use the syntax using .Gadfly or import .Gadfly, rather than the usual syntax. Otherwise you will get a warning about Gadfly not being in dependencies.

either way, i'd suggest creating an issue over at Requires.jl which describes this problem.

i also went looking for an Immerse PR by you but couldn't find one.

**Comment by jonathanBieler** [comment]
@bjarthur I haven't made a PR yet, still have some issues with transitioning to the new iteration protocole:

https://github.com/JuliaGraphics/Immerse.jl/blob/master/src/compose.jl#L286
